src/chain.js

```diff
@@ -2,14 +2,10 @@ const http = require('http');
 const { getBasic } = require('./utils/get_basic');
 
 /**
- * @param {http.ClientRequest} request
- * @param {net.Socket} source
- * @param {buffer.Buffer} head
- * @param {*} handlerOpts
- * @param {*} server
+ * @param {*} options
  */
-const chain = (request, source, head, handlerOpts, server) => {
-    if (head.length > 0) {
+const chain = ({ request, source, head, handlerOpts, server, plain }) => {
```

Comment:
Tiny thing: If `plain` is a boolean, in the Apify codebase by convention we prefix those variables with `has` or `is`. E.g. `isPlain`

Reply:
Good idea.